presentation/src/main/java/bg/government/virusafe/mvvm/fragment/AbstractFragment.kt

```diff
@@ -270,6 +274,23 @@ abstract class AbstractFragment<B : ViewDataBinding, VM : AbstractViewModel> :
 		}
 	}
 
+	fun showAgreementsDialog(
+		title: String,
+		description: String,
+		agreement: Agreement,
+		showAgreeBtn: Boolean = false,
+		onClick: OnDialogButtonListener? = null
+	) {
+		val dialog = AgreementsDialog.newInstance(title, description, agreement, showAgreeBtn)
+		val fm: FragmentManager? = activity?.supportFragmentManager
```

Comment:
You can call the let function on this line and create the dialog in it and show it.